postprocess_segmentation.py

```diff
@@ -498,12 +524,77 @@ def main(params):
+            if 'checkpoint' in tags.keys():
+                checkpoint_path = tags['checkpoint']
+                # add more info on a new layer waiting for Fiona 1.9
+                logging.info(f"\nAdding a layer 'extent_info' in {out_gen} with addintional informations.")
+                # Create a schema to store extent informations in the gpks
+                extent_schema_list = [("stac_item", 'str'), ("gdl_image", 'str')]
+                # list all layer(s) in the gpkg
+                layers = fiona.listlayers(out_gen)
+                # Look which class is use first by looking in the inf_outpath 
+                cls_name = next((elem for elem in classes_dict if elem in str(inf_outpath_ori)), None)  # TODO something when its None
+                # Check if the layer already created
+                if 'extent_info' in layers:
+                    # open the gpkg
+                    _gpkg = fiona.open(out_gen, layer='extent_info')
+                    new_extent_schema = _gpkg.schema.copy()
+                    # Add the model class name to the schema to fit what will be added
+                    new_extent_schema['properties'].update({"model_"+cls_name : 'str'})
+                    # extract information in the layer 
+                    prop = list(_gpkg.filter())[0]['properties']
+                    p = prop.copy()
+                    # close the gpkg to be able to rewrite it
+                    _gpkg.close()
+                    # add infrmation in the layer already there
+                    with fiona.open(returned_vector_pred, 'w', crs=_gpkg.crs, layer='extent_info', schema=new_extent_schema, driver='GPKG', overwrite=True) as ds:
+                        # add the checkpoint path associated with the model use for that class
+                        p.update({"model_"+cls_name : checkpoint_path})
+                        ds.write({'properties': p})
+                # If not there create the layer for extent information
+                else:
+                    # open the gpkg
+                    _gpkg = fiona.open(out_gen)
+                    # Add the model class name to the schema to fit what will be added
+                    extent_schema_list.append(("model_"+cls_name, 'str'))
+                    extent_schema = {'properties': OrderedDict(extent_schema_list)}
+                    # Create a new layer in the gpkg with all information on stac item, gdl image and model class
+                    with fiona.open(returned_vector_pred, 'w', crs=_gpkg.crs, layer='extent_info', schema=extent_schema, driver='GPKG') as ds:
+                        p = {
+                            "stac_item": item_url,
+                            "gdl_image": reg_cont_image,
+                        }
+                        # add the checkpoint path associated with the model use for that class
+                        p.update({"model_"+cls_name : checkpoint_path})
+                        ds.write({'properties': p})
+            
```

Comment:
This is also a lengthy piece of code, could this become a function of its own? The dosctrings of this potential function would help quickly understand what's going on here and what are inputs/outputs.